### DISCUSSION TITLE
请教各位大拿，如何在RequestConfig中，挂起请求，目的是实现双token的无感知刷新，完成用户认证，有没有实际的例子呢

### DISCUSSION THREAD

**Discussion by 262146150** [umijs/umi#11083] [Q&A]
请教各位大拿，如何在RequestConfig中，挂起请求，目的是实现双token的无感知刷新，完成用户认证，有没有实际的例子呢

**Comment by fz6m** [comment]
现在推荐自己配 axios 请求拦截器，顶层用 [react-query](https://umijs.org/docs/max/react-query) ，不要用 use request 了。

**Reply by CurlyJun1123** [reply]
有没有例子呢，我这边也有这样的一个需求，responseInterceptors 不能写异步请求

**Reply by fz6m** [reply]
如果改造影响比较大，给请求方法再封装一层就行了，里面对响应做一次检查，如果掉权了就去获取新的 token ，再重新请求就好。

**Comment by zengxpang** [comment]
有方案的话踢我下，现在
![361699869579_ pic](https://github.com/umijs/umi/assets/88088364/aeda9d16-bb1a-4af6-a65a-0b8734d26917)
接口报401的时候，可以请求到refresh接口拿到新的token，但是之前那个请求不会重新触发

**Comment by MeetYifan** [comment]
我使用的 umijs/max@4，看是否可以参考：

1. 引入依赖

<img width="1496" height="368" alt="image" src="https://github.com/user-attachments/assets/0922a4cb-de62-4ccf-b283-26994d8e8353" />

2. 响应拦截逻辑

<img width="1264" height="1394" alt="1753180451552" src="https://github.com/user-attachments/assets/927771d6-58e0-4c2b-b447-753a6733ec9c" />

3. token 失效尝试刷新 token

<img width="1496" height="1508" alt="1753180239902" src="https://github.com/user-attachments/assets/81bfdf3c-9429-43bb-8061-b4da4e981ca8" />
